data_structures/graph.ts

```diff
@@ -0,0 +1,109 @@
+class Graph {
+  constructor(amountOfVertices) {
+    this.vertices = [];
+    this.addVertices(amountOfVertices);
+  }
+
+  addVertices(amountOfVertices) {
+    for (let i = 0; i < amountOfVertices; i++) {
+      this.vertices.push([]);
+    }
+  }
+
+  addEdge(start, end) {
+    for(let i = 0; i < end; i++) {
+      this.vertices[start][i] = this.vertices[start][i] || 0;
+    }
+    this.vertices[start][end] = 1;
+  }
+
+  buildMatrix() {
+    this.matrix = [];
+    let maxEdgesLength = 0;
+    this.vertices.forEach(element => {
+      if (element.length > maxEdgesLength) {
+        maxEdgesLength = element.length;
+      }
+    });
+
+    for (let index = 0; index < this.vertices.length; index++) {
+      if (this.vertices[index].length < maxEdgesLength) {
+        const diff = maxEdgesLength - this.vertices[index].length;
+        this.vertices[index] = this.vertices[index].concat(Array(diff).fill(0));
+      }
+
+      this.matrix[index] = this.vertices[index];
+    }
+  }
+
+  getMatrix() {
+    return this.matrix;
```

Comment:
@dianalemen , this code will not compile. You do not have a field matrix in the graph class